bridge/opencensus/bridge.go

```diff
@@ -0,0 +1,222 @@
+	return parent
+}
+
+type span struct {
+	// We can't implement the unexported functions, so add the interface here.
+	octrace.Span
+	otSpan otel.Span
+}
+
+func (s *span) IsRecordingEvents() bool {
+	return s.otSpan.IsRecording()
+}
+
+func (s *span) End() {
+	s.otSpan.End()
+}
+
+func (s *span) SpanContext() octrace.SpanContext {
+	return otelSpanContextToOc(s.otSpan.SpanContext())
+}
+
+func (s *span) SetName(name string) {
+	s.otSpan.SetName(name)
+}
+
+func (s *span) SetStatus(status octrace.Status) {
+	s.otSpan.SetStatus(codes.Code(status.Code), status.Message)
+}
+
+func (s *span) AddAttributes(attributes ...octrace.Attribute) {
+	s.otSpan.SetAttributes(convertAttributes(attributes)...)
+}
+
+func convertAttributes(attributes []octrace.Attribute) []label.KeyValue {
+	otAttributes := make([]label.KeyValue, len(attributes))
+	for i, a := range attributes {
+		otAttributes[i] = label.KeyValue{
+			Key:   label.Key(a.Key()),
+			Value: convertValue(a.Value()),
+		}
+	}
+	return otAttributes
+}
+
+func convertValue(ocval interface{}) label.Value {
+	switch v := ocval.(type) {
+	case bool:
+		return label.BoolValue(v)
+	case int64:
+		return label.Int64Value(v)
+	case float64:
+		return label.Float64Value(v)
+	case string:
+		return label.StringValue(v)
+	default:
+		return label.StringValue("unknown")
+	}
+}
+
+func (s *span) Annotate(attributes []octrace.Attribute, str string) {
+	s.otSpan.AddEvent(str, otel.WithAttributes(convertAttributes(attributes)...))
+}
+
+func (s *span) Annotatef(attributes []octrace.Attribute, format string, a ...interface{}) {
+	s.Annotate(attributes, fmt.Sprintf(format, a...))
+}
+
+var (
+	uncompressedKey = label.Key("uncompressed byte size")
+	compressedKey   = label.Key("compressed byte size")
+)
+
+func (s *span) AddMessageSendEvent(messageID, uncompressedByteSize, compressedByteSize int64) {
+	s.otSpan.AddEvent("message send",
+		otel.WithAttributes(
+			label.KeyValue{
+				Key:   uncompressedKey,
+				Value: label.Int64Value(uncompressedByteSize),
+			},
+			label.KeyValue{
+				Key:   compressedKey,
+				Value: label.Int64Value(compressedByteSize),
+			}),
+	)
+}
+
+func (s *span) AddMessageReceiveEvent(messageID, uncompressedByteSize, compressedByteSize int64) {
+	s.otSpan.AddEvent("message receive",
+		otel.WithAttributes(
+			label.KeyValue{
+				Key:   uncompressedKey,
+				Value: label.Int64Value(uncompressedByteSize),
+			},
+			label.KeyValue{
+				Key:   compressedKey,
+				Value: label.Int64Value(compressedByteSize),
+			}),
+	)
+}
+
+func (s *span) AddLink(l octrace.Link) {
+	global.Handle(fmt.Errorf("ignoring OpenCensus link %+v for span %q because OpenTelemetry doesn't support setting links after creation", l, s.String()))
+}
+
+func (s *span) String() string {
+	return fmt.Sprintf("span %s",
+		s.otSpan.SpanContext().SpanID.String())
+}
+
+func otelSpanContextToOc(sc otel.SpanContext) octrace.SpanContext {
+	if sc.IsDebug() || sc.IsDeferred() {
+		// OTel don't support these options
```

Comment:
```suggestion
		// OpenCensus don't support these options
```

Reply:
I removed the comment since it is a duplicate of the error message anyways.